src/EncryptedRow.php

```diff
@@ -482,14 +491,16 @@ public function encryptRow(
             );
             if (
                 !empty($this->aadSourceField[$field])
-                    &&
+                &&
                 \array_key_exists($this->aadSourceField[$field], $row)
             ) {
-                $aad = (string) $row[$this->aadSourceField[$field]];
+                $aad = (string)$row[$this->aadSourceField[$field]];
             } else {
                 $aad = '';
             }
             if ($type === Constants::TYPE_JSON && !empty($this->jsonMaps[$field])) {
+                //decode json field then to take key from it to encrypt it
+                $row[$field] = isset($row[$field]) ? (array)json_decode($row[$field]) : [];
```

Comment:
This should probably be done at a higher level than inside our encryptRow() method.

Reply:
That was done also in the last commit with a new method to format the json field then return it to be used inside the encryptRow() method.